cwms-data-api/src/main/java/cwms/cda/api/TimeSeriesController.java

```diff
@@ -411,6 +378,24 @@ public void getAll(@NotNull Context ctx) {
             String begin = ctx.queryParam(BEGIN);
             String end = ctx.queryParam(END);
             String timezone = ctx.queryParamAsClass(TIMEZONE, String.class).getOrDefault("UTC");
+            String versionDateParam = ctx.queryParam(VERSION_DATE);
+            Timestamp versionDate = null;
+            if (versionDateParam != null) {
+                versionDate = Timestamp.from(DateUtils.parseUserDate(versionDateParam, timezone).toInstant());
+            }
+
+            String versionType = queryParamAsClass(ctx,
+                    new String[]{VERSION_TYPE, },
+                    String.class, VersionType.MAX_AGGREGATE.getValue(), metrics,
+                    name(TimeSeriesController.class.getName(), GET_ONE));
+
+            // Throw error if max aggregate version type is requested with a version date
+            // or single version version type is requested without a version date
+            if( (Objects.equals(versionType, "MAX_AGGREGATE") && !Objects.equals(versionDate, null))
+                || (Objects.equals(versionType, "SINGLE_VERSION") && Objects.equals(versionDate, null)) ) {
+                throw new IllegalArgumentException();
```

Comment:
Needs explinatory text. Additionally the exception handling in ApiServlet.java, and make sure it's going to map properly to one of the error codes. For example this would be 400 BAD REQUEST ... and then why.

Reply:
By explanatory text do you mean in the error message? I added a comment but I'm assuming you want a message that explains the relationship between versionDate and versionType

Reply:
Done